source/common/statistic_impl.cc

```diff
@@ -104,6 +106,38 @@ StatisticPtr SimpleStatistic::combine(const Statistic& statistic) const {
   return combined;
 }
 
+absl::StatusOr<std::unique_ptr<std::istream>> SimpleStatistic::serializeNative() const {
```

Comment:
when will this function return absl::Status?

Reply:
Fixed in f7b1269b

Reply:
Some statistics that don't implement this return `absl::Status(absl::StatusCode::kUnimplemented, "serializeNative not implemented.");
}`. 
Also the Hdr variant has:
```c++
absl::StatusOr<std::unique_ptr<std::istream>> HdrStatistic::serializeNative() const {
  char* data;
  if (hdr_log_encode(histogram_, &data) == 0) {
    auto write_stream = std::make_unique<std::stringstream>();
    *write_stream << absl::string_view(data, strlen(data));
    // Free the memory allocated by hrd_log_encode.
    free(data);
    return write_stream;
  }
  ENVOY_LOG(error, "Failed to write HdrHistogram data.");
  return absl::Status(absl::StatusCode::kInternal, "Failed to write HdrHistogram data");
}
```

In this case the simplicity is such that I don't see how this would need to return a status instead of a serialized representation; an exception is warranted in that case I guess.

Reply:
ah, thanks Otto for the explaination.